red/runtime/nodes/context/localfilesystem.js

```diff
@@ -203,8 +204,22 @@ LocalFileSystem.prototype.open = function(){
                     log.debug("Flushing localfilesystem context scope "+scope);
                     promises.push(fs.outputFile(storagePath + ".json", stringifiedContext.json, "utf8"));
                 });
-                delete self._pendingWriteTimeout;
-                return Promise.all(promises);
+                return Promise.all(promises).then(function(){
+                    if(Object.keys(self.pendingWrites).length > 0){
```

Comment:
I don't *think* this block is needed *if* we move the `delete self._pendingWriteTimeout` to the line above the `return Promise.all(...)`.

All of the code up to the return is synchronous - so we know there cannot be any additional calls to `set` up to that point. Clearing the timeout at that point will mean if a new write does come in whilst the file writes are happening, it will cue up a new timeout call.

Reply:
I have tested the changes I've described and they appear to work.

I may merge this PR as-is and then apply the changes in a subsequent commit - but will leave it for tonight.